molecule/base_molecule.h

```diff
@@ -383,6 +404,9 @@ namespace indigo
 
         Array<int> _hl_atoms;
         Array<int> _hl_bonds;
+        Array<int> _sl_atoms;
+        Array<int> _sl_bonds;
```

Comment:
Basically, they should be `Array<bool>`, since selection state can be only: 0/1 or false/true, or unselected/selected.
At least, this would save memory twice. 
The same refers to old `_hl_atoms` and `_hl_bonds`.
Of course, `class Array` is not perfect for `bool` type, but nothing limits us from using `std::vector<bool>` or (differently) `std::set<int>`, where `int` is `index`.

Reply:
It came from base_molecule.h where _hl_atoms and _hl_bonds were defined as:

Array<int> _hl_atoms;
Array<int> _hl_bonds;

Moving to std::vector and std::set should be made as a part of another PR related to refactoring and optimisations because there are plenty of places in Indigo where it's possible to see inefficient code like iterating thru arrays instead of using sets.

Reply:
agree that Array<bool> or Array<char> will save some memory. But in terms of performance it will be the same. Do not agree to move to std::vector (and especially set) - as it will decrease a performance. 
In this kind of changes please always consider that number of atoms and bonds is limited and memory optimization has no value.